server/mailet/rate-limiter/src/test/scala/org/apache/james/transport/mailets/PerRecipientRateLimitMailetIntegrationTest.scala

```diff
@@ -0,0 +1,278 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ * http://www.apache.org/licenses/LICENSE-2.0                   *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.mailets
+
+import java.io.File
+import java.util.concurrent.TimeUnit
+
+import org.apache.james.core.builder.MimeMessageBuilder
+import org.apache.james.mailbox.model.{MultimailboxesSearchQuery, SearchQuery}
+import org.apache.james.mailets.TemporaryJamesServer
+import org.apache.james.mailets.configuration.Constants.{DEFAULT_DOMAIN, LOCALHOST_IP, PASSWORD}
+import org.apache.james.mailets.configuration.{Constants, MailetConfiguration, MailetContainer, ProcessorConfiguration}
+import org.apache.james.modules.MailboxProbeImpl
+import org.apache.james.modules.protocols.{ImapGuiceProbe, SmtpGuiceProbe}
+import org.apache.james.rate.limiter.memory.MemoryRateLimiterModule
+import org.apache.james.transport.mailets.PerRecipientRateLimitMailetIntegrationTest._
+import org.apache.james.transport.matchers.All
+import org.apache.james.utils.{DataProbeImpl, SMTPMessageSender, SpoolerProbe, TestIMAPClient}
+import org.apache.mailet.base.test.FakeMail
+import org.assertj.core.api.AssertionsForClassTypes.assertThat
+import org.assertj.core.api.SoftAssertions
+import org.junit.jupiter.api.io.TempDir
+import org.junit.jupiter.api.{AfterEach, BeforeEach, Test};
+
+object PerRecipientRateLimitMailetIntegrationTest {
+  val RECIPIENT: String = "recipient@" + DEFAULT_DOMAIN
+  val RECIPIENT2: String = "recipient2@" + DEFAULT_DOMAIN
+  val SENDER: String = "sender@" + DEFAULT_DOMAIN
+  val SENDER2: String = "sender2@" + DEFAULT_DOMAIN
+  val RATE_LIMIT_STATE: String = "rate_limit_state"
+}
+
+class PerRecipientRateLimitMailetIntegrationTest {
```

Comment:
```suggestion
class PerRecipientRateLimitMailetTest {
```

In this maven project, we SHOULD have a unit test... 

Integration tests should be in `/server/mailet/intergation-tests/`